### PR TITLE
feat: add setHeader public api

### PR DESCRIPTION
**Pull request by seonim-ryu**
### Please check if the PR fulfills these requirements
- [x] It's submitted to right branch according to our branching model
- [x] It's right issue type on title
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] It does not introduce a breaking change or has description for the breaking change

### Description

* Needs to dynamically change the complex column in header.
* Add `setHeader` api.
* test code
```js
grid.setHeader({
  complexColumns: [
    {
      title: 'Basic',
      name: 'mergeColumn1',
      childNames: ['name', 'artist']
    },
    {
      title: 'Extra',
      name: 'mergeColumn2',
      childNames: ['type', 'release', 'genre']
    },
    {
      title: 'Detail',
      name: 'mergeColumn3',
      childNames: ['mergeColumn1', 'mergeColumn2']
    },
    {
      title: 'Count',
      name: 'mergeColumn4',
      childNames: ['downloadCount', 'listenCount']
    },
    {
      title: 'Album Info1',
      name: 'mergeColumn5',
      childNames: ['mergeColumn4']
    }
  ]
});
```

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

File: src/js/grid.js [hunk 1]
```diff
@@ -730,6 +730,26 @@ var Grid = View.extend(/** @lends Grid.prototype */{
...
+        if (options.complexColumns) {
+            this.modelManager.columnModel.set({
+                complexHeaderColumns: options.complexColumns
+            }, {
+                silent: !!options.height
+            });
+        }
+
+        if (options.height) {
+            this.modelManager.dimensionModel.set('headerHeight', options.height);
```
Comment (seonim-ryu) [inline src/js/grid.js]: @todo `options.height` 변수 캐싱으로 변경

Comment (dongwoo-kim) [reply src/js/grid.js]: 이렇게 구현하면 `options.height`가 기존 값이랑 동일한 경우 `complexColumns`가 View에 반영이 안되는 문제가 있을 것 같습니다.

Comment (seonim-ryu) [reply src/js/grid.js]: @dongwoo-kim 
리뷰 반영했습니다. 근데 뷰에서 모델이 점점 늘어나는 구조는 아름답지 않군요 😥

**Comment by jung-han** [comment]
ok to test

**Pull request merged**